File: Python_Design-Patterns/Taller_Adapter/WheelsAdapter.py
from Taller_Adapter.MedioTransporte import MedioTransporte
from Taller_Adapter.Wheels import Wheels


class WheelsAdapter(MedioTransporte):

    def __init__(self):
        super().__init__()
        self._wheels = Wheels()

    @property
    def costo(self):
        return self._wheels.precio

    @costo.setter
    def costo(self, costo):
        self._wheels.precio = costo

    @property
    def horario(self):
        return self._wheels.hora

    @horario.setter
    def horario(self, horario):
        self._wheels.hora = horario

File: Java_Design-Patterns/src/Taller_Adapter/WheelsAdapter.java
package Taller_Adapter;

public class WheelsAdapter extends MedioTransporte{

    private Wheels wheels;

    public WheelsAdapter() {
        super();
        this.wheels = new Wheels();
        System.out.println("Adaptando Wheels");
    }

    public double getCosto() {
        return wheels.getPrecio();
    }

    public void setCosto(double costo) {
        wheels.setPrecio(costo);
    }

    public String getHorario() {
        return wheels.getHoras();
    }

    public void setHorario(String horario) {
        wheels.setHoras(horario);
    }
}
